cEP-0029.md

```diff
@@ -0,0 +1,63 @@
+# Gitmate for coala
+
+| Metadata |                                                       |
+| -------- | ----------------------------------------------------- |
+| cEP      | 29                                                    |
+| Version  | 1.0                                                   |
+| Title    | Gitmate for coala                                     |
+| Authors  | Sanchit Gupta <mailto:sanchitgupta.072@gmail.com>     |
+| Status   | Implementation Due                                    |
+| Type     | Process                                               |
+
+## Abstract
+
+This cEP describes the improvements to IGitt and Gitmate, which include
+upgrading dependencies, making it functional, changes to plugins, etc. as a part
+of the [GSoC project](https://summerofcode.withgoogle.com/projects/#6263057774280704).
+
+## Introduction
+
+Gitmate, a git workflow management bot created by coala, hasn't been actively maintained in the last 3 years. My project aims to update it and bring it back to active use for the coala community. I plan to undertake various improvements to Gitmate as well as IGitt, the underlying library that powers Gitmate's git operations.
+
+## Upgrade Dependencies
+
+IGitt and Gitmate are both severly outdated in terms of dependencies. Support needs to be added for the latest versions of python, and most packages need to be updated to their latest versions. IGitt is currently using Jira API v2 which can be upgraded to v3. Currently, Angular is running v4. It needs fundamental changes to upgrade it to v12.
```

Comment:
Moving from angular v4 to v12 is a major change. According to https://update.angular.io/?v=4.0-12.0, looks like it's a big change, there should be milestone around it. Can you please make one

Reply:
Yes, noted!

Reply:
Yup, let's also plan a feasible time period in which this can be done. You can try to do a part of this as a part of this period and see if you're able to migrate some things. This will give us a rough timeline in which we can finish this up.